textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java

```diff
@@ -53,6 +57,9 @@ public DictionaryMatcher(IPredicate predicate) {
     @Override
     public void open() throws DataFlowException {
         try {
+        	cursor = 0;
+        	limit = Integer.MAX_VALUE;

```

Comment:
Let `DictionaryMatcher dictionaryMatcher = new DictionaryMatcher(dictPredicate);`. Then if `dictionaryMatcher.setLimit();` is called and then `dictionaryMatcher.open();` is called, whatever limit is set is overwritten to `Integer.MAX_VALUE` isn't it?